torch/nn/modules/module.py

```diff
@@ -2101,6 +2114,13 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
             If :attr:`assign` is ``True`` the optimizer must be created after
             the call to :attr:`load_state_dict`.
 
+        .. warning::
```

Comment:
This behavior differs from the default/might be surprising, so I thought it might be worth mentioning.

It is probably unlikely that the user will use the state_dict after loading though, so maybe it is okay to omit this

Reply:
The default is to .copy_() into what is here right now. So yes the new behavior differs from it but that is much broader than what you mention here.
Also I think the flag setter is the right place to explain what the flag does.